### PR TITLE
Add Test and Lint

### PR DESCRIPTION
**Pull request by rthor**
Adds the `jest` testing framework as well as `tslint` (eslint for typescript). Also runs those on CircleCI.

### CODE REVIEW

**Comment by rthor** [comment]
Might be nice to squash this when you merge.

**Comment by rthor** [comment]
Oh, and FYI, the actual unit test is just there to test the system. Not meant as an actual test.

File: packages/tux/src/components/TuxProvider/TuxProvider.tsx [hunk 1]
```diff
@@ -4,8 +4,8 @@ import ModalContainer, { openModal } from '../TuxModalContainer'
 import TuxBar from '../TuxBar'
 import TuxModal from '../TuxModal'
 
-export interface Props {
-  adapter: Object
+interface Props {
```
Comment (eirikurn) [inline packages/tux/src/components/TuxProvider/TuxProvider.tsx:7]: Why no export. Thought it was needed for compiling library definitions.

File: .gitignore [hunk 2]
```diff
@@ -3,7 +3,7 @@ npm-debug.log
 lerna-debug.log
 
 # production
-packages/*/lib
+packages/**/lib
```
Comment (eirikurn) [inline .gitignore:6]: Will we have deeper lib folders?

File: packages/tux-adapter-contentful/package.json [hunk 3]
```diff
@@ -14,5 +15,16 @@
...
+  },
+  "jest": {
+    "transform": {
+      ".(ts|tsx)": "<rootDir>/../../node_modules/ts-jest/preprocessor.js"
+    },
+    "testRegex": "\\.(test|spec)\\.(ts|tsx)$",
+    "moduleFileExtensions": [
+      "ts",
+      "tsx",
+      "js"
```
Comment (eirikurn) [inline packages/tux-adapter-contentful/package.json:27]: Is this needed for node_modules?

**Review by eirikurn — APPROVED** [review]

**Comment by eirikurn** [comment]
Awesome

File: packages/tux/src/components/TuxProvider/TuxProvider.tsx (hunk 1 above)
Comment (rthor) [reply packages/tux/src/components/TuxProvider/TuxProvider.tsx:7]: Oh, my bad... Had a merge conflict there and just missed the export keyword.

File: .gitignore (hunk 2 above)
Comment (rthor) [reply .gitignore:6]: The single star glob didn't work for me, git always tried to add new files... Maybe I have an outdated git version 🤔

File: packages/tux-adapter-contentful/package.json (hunk 3 above)
Comment (rthor) [reply packages/tux-adapter-contentful/package.json:27]: Doesn't build without it.

**Pull request merged**